parachain-tracer/src/tracker.rs

```diff
@@ -304,6 +310,13 @@ impl ParachainBlockTracker for SubxtTracker {
 					}
 					self.stats.on_backed();
 					metrics.on_backed(self.para_id);
+
+					if self.handle_on_demand_delay("backed", metrics) {
+						self.on_demand_order_block = None;
```

Comment:
this could be moved in `handle_on_demand_delay`. Ideally there is only one place where you mutate these `on_demand_*` variables, otherwise things get really messy. 

Overall readability is not that great as we add more functionality.

I'd love to see a followup refactoring PR.